### ISSUE TITLE
[Bug]: wasm js bindings generated by wasm-pack is not compatible with raw browser import

### ISSUE THREAD

**Issue by wighawag** [dojoengine/dojo.js#268]
### Dojo.js Version

all

### What happened?

Currently dojo.js use [wasm-pack](https://rustwasm.github.io/wasm-pack/) to generate the js boilerplate to load wasm in browser/node/etc...

Unfortunately wasm-pack generate js file that load wasm via es import (`import * as wasm from "./<lib>.wasm"`) which is not support by browser.

Some bundler will magically transform such import so browser can import it, some other (like vite) requires a community plugin to make it work. But ideally dojoengine should work without such external tool and as such it would be great if dojo.js output a js boilerplate compatible with browser (without require any other toool)

It seems using the [wasm-bindgen cli](https://rustwasm.github.io/wasm-bindgen/) to generate the js binding generate the correct js so it can be loaded in browser without any extra bundler

### What is the expected behavior?

I expect to be able to use dojo with raw browser import, without the need for a bundler

### Steps to reproduce

generate import map for torii-client and its dependencies

set them up in the index.html and add a script that load the torii-client and you ll'see the following error:

```
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/wasm". Strict MIME type checking is enforced for module scripts per HTML spec.
```

see reproduction repo : https://github.com/bug-reproduction/torii-wasm-in-browser

see codepen : https://codepen.io/wighawag/pen/RwzddBM

### Is there an existing issue for this?

- [X] I have searched the existing issues and verified no issue exits for this problem.

**Comment by ShantelPeters** [comment]
Hi @wighawag please can I take this up

**Comment by wighawag** [comment]
@ShantelPeters  I am not the maintainer of this repo

**Comment by ShantelPeters** [comment]
Alright thank you @wighawag

**Comment by wighawag** [comment]
Linking the reply here : https://github.com/dojoengine/dojo.c/issues/55#issuecomment-2351789568

It suggest to use `--target no-modules` but this is less than ideal. we should be able to use es module in modern browser. And I believe this require some more wiring on using it. though this might be always the case.

**Comment by ponderingdemocritus** [comment]
yes, these wasm modules are annoying. I'll try and support this soon - or feel free to open PR.
